File: fixtures/go/animal/animal.go
package animal

import "fmt"

// Animal 动物接口
type Animal interface {
	MakeSound()
	Eat(food string)
	GetName() string
	GetAge() int
}

// BaseAnimal 动物基类
type BaseAnimal struct {
	Name string
	Age  int
}

// MakeSound 发出声音
func (a *BaseAnimal) MakeSound() {
	fmt.Println("Some generic animal sound")
}

// Eat 吃东西
func (a *BaseAnimal) Eat(food string) {
	fmt.Printf("%s is eating %s\n", a.Name, food)
}

// GetName 获取名字
func (a *BaseAnimal) GetName() string {
	return a.Name
}

// GetAge 获取年龄
func (a *BaseAnimal) GetAge() int {
	return a.Age
}

File: fixtures/rust/animal.rs
pub trait Animal {
    fn make_sound(&self);
    fn eat(&self, food: &str);
    fn get_name(&self) -> &str;
    fn get_age(&self) -> i32;
}

pub struct BaseAnimal {
    pub name: String,
    pub age: i32,
}

impl BaseAnimal {
    pub fn new(name: &str, age: i32) -> Self {
        BaseAnimal {
            name: name.to_string(),
            age,
        }
    }
}

impl Animal for BaseAnimal {
    fn make_sound(&self) {
        println!("Some generic animal sound");
    }
    
    fn eat(&self, food: &str) {
        println!("{} is eating {}", self.name, food);
    }
    
    fn get_name(&self) -> &str {
        &self.name
    }
    
    fn get_age(&self) -> i32 {
        self.age
    }
}
